Fabric.Terminology.API/Scripts/Terminology-Install-Utilities.psm1

```diff
@@ -386,6 +386,174 @@ function Publish-TerminologyDacpac() {
+            $terminologyResponse = Invoke-RestMethod -Uri $terminologyUrl -Method GET -UseDefaultCredentials
+        }
+        if($sharedTerminologyResponse -ne "Failed") {
+            $sharedTerminologyResponse = Invoke-RestMethod -Uri $terminologyUrl -Method GET -UseDefaultCredentials
+        }
+
+        if($terminologyResponse.Status -eq "Failed" -or $sharedTerminologyResponse.Status -eq "Failed") {
+            Write-DosMessage -Level "Error" -Error "One of the batch executions has failied. Please check EDW Console for additional logging. Halting the terminology registration"
+            break
+        }
+        if($terminologyResponse.Status -eq "Success" -and $sharedTerminologyResponse.Status -eq "Success") {
+            Write-DosMessage -Level "Error" -Message "Both batch executions have successfully been run."
+            $wasSuccessful = 1;
+            break
+        }
+
+        Write-DosMessage -Level "Information" -Message  "Terminology status: $($terminologyResponse.Status)"
+        Write-DosMessage -Level "Information" -Message  "SharedTerminology status: $($sharedTerminologyResponse.Status)"
+    }
+    return $wasSuccessful;
+
+}
+
+
+function Add-MetadataAndStructures() {
+    param(
+        [PSCumstonObject] $Config
+    )
+    $discoveryServiceUrl = $Config.discoveryServiceUrl
+    $metadataDatabase = $Config.metadataDatabaseName
+    $sqlAddress = $Config.sqlAddress
+      
+   
+    # get IdentityID from IdentityBASE
+    $user = "$env:USERDOMAIN\$env:USERNAME"
+    $identityBaseQuery = "SELECT TOP (1) [IdentityID] FROM [$metadataDatabase].[CatalystAdmin].[IdentityBASE] WHERE UPPER(IdentityNM) = UPPER('$user')"
+    $identityQueryResult = Invoke-SqlCommand -SqlServerAddress $sqlAddress -DatabaseName $metadataDatabase -Query $identityBaseQuery
+    $identityId = $identityQueryResult.Table.IdentityID
+
+    if ($identityId) {
+        Write-DosMessage -Level "Information" -Message  "IdentityID for $user`: $identityId"
+    }
+    else {
+        Write-DosMessage -Level "Information" -Message   "User is not in IdentityBASE... Adding user: $user"
+        Invoke-Sql -sqlCommand "INSERT INTO [$metadataDatabase].[CatalystAdmin].[IdentityBASE] (IdentityNM) VALUES (UPPER('$user'))";
+        $identityBaseQuery = "SELECT TOP (1) [IdentityID] FROM [$metadataDatabase].[CatalystAdmin].[IdentityBASE] WHERE UPPER(IdentityNM) = UPPER('$user')"
+        $identityQueryResult = Invoke-SqlCommand -SqlServerAddress $sqlAddress -DatabaseName $metadataDatabase -Query $identityBaseQuery
+        $identityId = $identityQueryResult.Table.IdentityID
+    }
+
+    $dataProcessingServiceRoleId = Get-RoleId -role "DataProcessingServiceUser"
+
+    # POST Terminology data marts to MDS
+    Write-DosMessage -Level "Information" -Message   "Creating Terminology metadata. This could take upwards of ten minutes depending on the availability and capacity of the server. In most cases it takes between 3 and 6 minutes."
```

Comment:
Nice, good job letting the user know

Reply:
I would be careful giving a time estimate, although you may be right most of the time. It will really frustrate and confuse users when you are wrong